### ISSUE TITLE
[FEATURE REQUEST] Add support for YAML sequences

### ISSUE THREAD

**Issue by yantosca** [yantosca/qfyaml#1]
At present, qfyaml cannot handle YAML sequences, i.e.
```
variable:
 - 1.0
 - 2.0
 - 3.0
```
etc.

As time allows it would be good to add this as a future enhancement.

**Comment by yantosca** [comment]
This is mostly resolved by commits fdb76c6ce23cb63af927e292595fef26a57d558f and cd714e1324ca84d2ad6aa3aba2f8c7bc0c9e527d. 

The test program `src/test_qfyaml.F90` can now parse this YAML file:
```yaml
---
#### input.yml - Input for test_qfyaml.F90

#### Test going back & forth between levels
weather:
  humidity: 99.85858265
  temperature:
    daily: 23.0434356
      weekly:
        units: K
  pressure: 1013.25

#### Test parsing different types of input
author:
  age: 29
  fav_reals: [1.0, 2.0]
  more_reals: 3.14159 2.78128 8.5739 101.325
  lots_of_work: false

#### Testing YAML sequences with various nesting levels

fruits:
  - Apples
  - Bananas
  - Oranges

more_fruits:
  p_fruits:
    - Pears
    - Plums
    - Peaches
    - Pomegranites

even_more_fruits:
  exotic_fruits:
    hard_to_find:
      - Kumquats
      - Kiwi
      - Passion_fruit     # NOTE: use underscores for multiple words,
      - Star_fruit        # as there an issue in parsing spaces in YAML
      - Durians           # sequences, to be addressed in future.

#### Test parsing a variable without a category
filename: another_file

#### Test parsing strings
author_name:
  first: Homer
  full: Homer J. Simpson
```
and return this output:
```console
### Reading input.yml

### YAML VARIABLES
author%age                       :      29
author%fav_reals                 :    1.00   2.00
author%more_reals                :    3.141590   2.781280   8.573900 101.324997
author%lots_of_work              :       F
author_name%first                : Homer
author_name%full                 : Homer J. Simpson
filename                         : another_file
weather%humidity                 :     99.858582
weather%temperature%daily        :     23.043436
weather%temperature%weekly%units : K
weather%pressure                 :   1013.250000

### YAML SEQUENCES
fruits
           1 Apples
           2 Bananas
           3 Oranges

more_fruits%p_fruits
           1 Pears
           2 Plums
           3 Peaches
           4 Pomegranites

even_more_fruits%exotic_fruits%hard_to_find
           1 Kumquats
           2 Kiwi
           3 Passion_fruit
           4 Star_fruit
           5 Durians

#### finishing
```
One issue yet to be resolved is that YAML sequence items with spaces cannot be read as a single item.  For example
```yaml
- Passion fruit
```
would be not parsed as a single item but as two separate items `Passion` and 'fruit`.  To get around this behavior, we recommend using underscores for YAML sequence items.

A fix will be forthcoming sometime soon.

**Comment by yantosca** [comment]
We will close this issue.  A new issue #5 has been opened to track the fix for YAML sequences not parsing sequence items separated by spaces.
